src/stream.rs

```diff
@@ -426,8 +431,33 @@ pub(crate) fn connect_host(unit: &Unit, hostname: &str, port: u16) -> Result<Tcp
         stream.set_write_timeout(unit.agent.config.timeout_write)?;
     }
 
-    if proto == Some(Proto::HTTPConnect) {
-        if let Some(ref proxy) = proxy {
+    if proto == Some(Proto::HTTPSConnect) || proto == Some(Proto::HTTPConnect) {
+        if let Some((mut stream, proxy)) = match (proto, proxy) {
+            (Some(Proto::HTTPSConnect), Some(ref proxy)) => {
+                let tls_conf;
+                #[cfg(feature = "native-tls")]
+                {
+                    tls_conf = native_tls::TlsConnector::new().unwrap();
+                }
+                #[cfg(not(feature = "native-tls"))]
+                {
+                    tls_conf = default_tls_config();
+                }
+                let proxy_conn = tls_conf
+                    .connect(&proxy.server, Stream::from_tcp_stream(stream))
+                    .unwrap();
+                Some((Stream::new(proxy_conn), proxy))
+            }
+            (Some(Proto::HTTPConnect), Some(ref proxy)) => {
+                Some((Stream::from_tcp_stream(stream), proxy))
+            }
+            _ => {
+                return Err(Error::new(
+                    ErrorKind::ProxyConnect,
+                    Some("No proxy defined, but proto set".into()),
+                ));
+            }
+        } {
```

Comment:
To not lock us down to these two TLS implementations, this should use the TlsConnector configured in the Agent.

Reply:
Done